### PR TITLE
Move to single cache impl.

### PR DESCRIPTION
**Pull request by mikehelmick**
* Move secret manager cache to internal/cache write through impl
* Start caching HealthAuthoirty configuration in verificaiton code

/fixes 555

### CODE REVIEW

File: internal/verification/config.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// limitations under the License.
+
+package verification
+
+import "time"
+
+// Config represents the available configuration for the public health authority
+// verification piece.
+type Config struct {
+	CacheDuration time.Duration `env:"HEALTH_AUTHORITY_CACHE_DURATION, default=5m"`
```
Comment (sethvargo) [inline internal/verification/config.go:22]: Should this be `VERIFICATION_CACHE_DURATION`?

Comment (mikehelmick) [reply internal/verification/config.go]: done

**Comment by sethvargo** [comment]
@mikehelmick seems like a legit failure.

**Comment by mikehelmick** [comment]
> @mikehelmick seems like a legit failure.

yeah, looking.

**Comment by mikehelmick** [comment]
pushed fix - was indeed a legit bug

race condition that isn't always triggered by the test

**Comment by mikehelmick** [comment]
added test that always cause the race condition to be hit ⏲️

**Pull request merged**